driver/src/main/scala/api/MongoDriver.scala

```diff
@@ -301,22 +378,20 @@ class MongoDriver(
         ()
       }
 
-      /*
-      case CloseWithTimeout(timeout) =>
-        if (isEmpty) context.stop(self)
-        else context.become(closing(timeout))
-         */
-
       case Close => {
         logger.debug(s"[$supervisorName] Close the supervisor")
 
-        if (isEmpty) context.stop(self)
-        else context.become(closing(Duration.Zero))
+        if (isEmpty) {
+          context.stop(self)
+          sender() ! Closed
+        } else {
+          context.become(closing(mutable.Queue[ActorRef](sender)))
```

Comment:
I don't see the point for the function (see next comment about `closing`) creating the new `Receive` context to be passed the mutable queue.

As using such mutable queue is internal implementation details for such function, this one should be responsible for initializing it (maybe as `@volatile`).